src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala

```diff
@@ -0,0 +1,73 @@
+package uk.gov.nationalarchives.downloadfiles
+
+import java.net.URLDecoder
+import java.nio.file.Paths
+import java.util.UUID
+
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3EventNotificationRecord
+import com.typesafe.config.{Config, ConfigFactory}
+import graphql.codegen.GetOriginalPath.getOriginalPath.{Data, Variables, document}
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
+import sttp.client.{Identity, NothingT, SttpBackend}
+import uk.gov.nationalarchives.tdr.{GraphQLClient, GraphQlResponse}
+import uk.gov.nationalarchives.tdr.error.NotAuthorisedError
+import uk.gov.nationalarchives.tdr.keycloak.KeycloakUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class FileUtils()(implicit val executionContext: ExecutionContext) {
+  val config: Config = ConfigFactory.load
+
+  case class DownloadFilesException(msg: String) extends RuntimeException
+
+  implicit class OptFunction(strOpt: Option[String]) {
+    def failIfPathEmpty(fileId: UUID): Try[String] = if ((strOpt.isDefined && strOpt.get.isEmpty) || strOpt.isEmpty) {
```

Comment:
What do you think about this refactoring to make the logic a bit clearer?

```
strOpt match {
  case Some(path) if path.nonEmpty => Success(path)
  case _ => Failure(new RuntimeException(s"The original path for fileId $fileId is missing or empty"))
}
```

Reply:
Very nice :smile: I've changed it.1